### PR TITLE
Do Not Resuscitate Status, Mechanics, and SoP

### PR DESCRIPTION
**Pull request by matttheficus**
# SoP Stuff (SUBJECT TO CHANGE VIA ITERATION W/ DEV/ADMIN TEAM)

# !!!PLEASE DO NOT MERGE UNTIL HEADS CAN MERGE THE SOP STUFF!!!

### Do Not Resuscitate - Security (DNR-S)
1. All DNR-S notices must be authorized by a proper adjudicator. Any unauthorized DNR-S notices are to be considered Murder. If no authorized adjudicator is available as stated in these Standards, authorization must be given by Central Command.
2. DNR-S authorization for Prisoners who have confirmed via evidence to have committed a 5XX - Capital Crime and succumbed to injuries in conflict with Security must be given by the Magistrate. If there is no active Magistrate, the Captain may give authorization. In the events of a Code Red Alert and no active Captain, the Head of Security may authorize the notice.
3. DNR-S authorization for Prisoners who have escaped Permanent Confinement and Executed may be given by the Warden, Head of Security, Magistrate or Captain.
4. DNR-S authorization for Prisoners who have been Executed for all other legal reasons are authorized automatically by setting the Prisoner to Execute.
5. All DNR-S bodies must be placed in a Red DNR-S Bodybag. Upon storage, the name of the Prisoner and the approving adjudicator must be placed on the name tag.
6. The revival of any individual who is denoted as DNR-S status is considered Grand Sabotage.

### Do Not Resuscitate - Medical (DNR-M)
1. All DNR-M notices must be authorized by the proper adjudicator. Any unauthorized DNR-M notices are to be considered Murder. If no authorized adjudicator is available as stated in these Standards, authorization must be given by Central Command.
2. DNR-M authorization for suicide victims may be approved by the Coroner. The Chief Medical Officer may, at their discretion, override this decision. In the absence of any authorized Medical staff, the Captain may approve the DNR-M notice instead.
3. All DNR-M bodies must be placed in a Blue DNR-M Bodybag. Upon storage, the name of the patient and the approving adjudicator must be placed on the name tag.
4. The revival of any individual who is denoted as DNR-M status is not considered a crime. However, as per the Chief Medical Officer, it can be treated as misappropriation of Medical Supplies and dealt with via reprimand, removal of Morgue access, or demotion.

### Kill on Sight (KoS) Orders
1. All KoS notice must be authorized by a proper adjudicator by setting their status as DNR-S, regardless of their current medical status. Any unauthorized KoS notices are to be considered Murder. If no authorized adjudicator is available as stated in these Standards, authorization must be given by Central Command.
2. A KoS notice may only be considered in the case active large-scale mass casualty events, the widespread and purposeful destruction of the Station or Emergency Escape Shuttle’s integrity, or any threat listed under “Kill on Sight” within the Space Law Lethal Force chart.
3. KoS authorization for individuals who have confirmed via evidence to have met the requirements above must be given by the Magistrate. If there is no active Magistrate, the Captain may give authorization. In the events of a Code Red Alert and no active Captain, the Head of Security may authorize the notice.

---

## What Does This PR Do
Adds mechanics and systems to allow for more easily tracked and adjudicated DNR Notices to be placed by Security/Medbay.

These include:
- Separate bodybags for Medical and Security DNR Notices, as per SoP. New Morgue Tray lights as well.
- Security HUD Status for DNR-S. DNR-S is automatically applied to Medical Records when Execute status is set.

## MMTODO
- [x] SecRecord Stuff + CSS
- [x] MedRecord Stuff + CSS
- [ ] Sec/Med Record Sync for Executions
- [ ] Bodybag Sprites
- [ ] SecHUD Icon
- [ ] Cargo Orderable Restocks for Bags and/or craftable with plastic

## Why It's Good For The Game
There is currently no real DNR SoP for the Crew. As such, some has been written up and approved provisionally, but needs some items added to the game.

## Images of changes
TBD

## Testing
TBD as I finish coding it up this weekend

## Changelog
:cl:
add: added DNR-S and DNR-M statuses and mechanics
add: added DNR-S and DNR-M bodbags
/:cl:

### CODE REVIEW

File: code/game/objects/items/bodybag.dm [hunk 1]
```diff
@@ -12,6 +12,28 @@
...
 	qdel(src)
 
+/obj/item/bodybag/medical_dnr
+	name = "body bag (medical DNR)"
+	desc = "A folded bag designed for the storage and transportation of cadavers under a Medical DNR Notice."
+	icon = 'icons/obj/bodybag.dmi'
+	icon_state = "MMTODO"
+
+/obj/item/bodybag/medical_dnr/attack_self(mob/user)
+	var/obj/structure/closet/body_bag/medical_dnr/R = new /obj/structure/closet/body_bag/medical_dnr(user.loc)
```
Comment (Contrabang) [inline code/game/objects/items/bodybag.dm:22]: This should be a var on `/obj/item/bodybag` so that attack_self doesn't have to be copied 3 times

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/game/objects/items/bodybag.dm (hunk 1 above)
Comment (matttheficus) [reply code/game/objects/items/bodybag.dm:22]: yeahhhhh I copy/pasted for proof of concept - Ill avoid coding warcrimes once i do a pass over with the sprites coming up :xdd:

File: code/game/data_huds.dm [hunk 2]
```diff
@@ -266,6 +266,9 @@
 				if(SEC_RECORD_STATUS_EXECUTE)
 					holder.icon_state = "hudexecute"
 					return
+				if(SEC_RECORD_STATUS_DNRS)
+					holder.icon_state = "MMTODO"
```
Comment (Sirryan2002) [inline code/game/data_huds.dm:270]: :O

Comment (Spaghetti-bit) [reply code/game/data_huds.dm:270]: :OOOO

**Comment by Daylight2** [comment]
Small nitpick on the SOP: Setting someone to KOS without proper authorization should probably be attempted murder not murder, unless the person actually dies as a result.

**Comment by matttheficus** [comment]
I currently dont have time to finish this, have some placeholder-esque sprites, and dont have the motivation to really do this in general with work + WoW xpac and some other games with IRL buddies.

Someone is fully free to take this PR if they so desire (I can give ya the sprites I do have) and/or go for the SoP changes alone in some form.

**Pull request closed without merging**